### PR TITLE
Add catalog for ukko

### PR DESCRIPTION
**Pull request by mgrabovsky**
I've tried to clean up and generalise the code a bit. It should be slightly more maintainable now, but it's still kind of a mess.

### CODE REVIEW

**Comment by darkengine-io** [comment]
This conflicted with upstream vichan so I couldn't merge it in. If you fix the post-merge conflicts and re-push, I will gladly accept it.

**Comment by mgrabovsky** [comment]
I've now rebased and modified the code. Everything seems to be working as before. If there's anything broken, hit me.

File: templates/themes/ukko/theme.php [hunk 1]
```diff
@@ -29,6 +29,7 @@ public function build($mod = false) {
 			$overflow = array();
 			$board = array(
 				'url' => $this->settings['uri'],
+				'uri' => $this->settings['uri'],
```
Comment (darkengine-io) [inline templates/themes/ukko/theme.php]: Does this need to be here?

File: templates/themes/catalog/info.php [hunk 2]
```diff
@@ -1,27 +1,27 @@
 <?php
 	$theme = array();
-	
+
```
Comment (darkengine-io) [inline templates/themes/catalog/info.php]: Weird whitespace things throughout this file

File: inc/functions.php [hunk 3]
```diff
@@ -1676,11 +1677,13 @@ function buildJavascript() {
 	}
 
 	$code_stylesheets = array();
-	foreach ($config['code_stylesheets'] as $name => $uri) {
-		$code_stylesheets[] = array(
-			'name' => addslashes($name),
-			'uri' => addslashes((!empty($uri) ? $config['uri_stylesheets'] : '') . $uri));
-	}
+	if (isset($config['code_stylesheets'])) {
```
Comment (darkengine-io) [inline inc/functions.php]: Is this supposed to be here?

**Comment by darkengine-io** [comment]
I think there are a few things that may have been added in by accident from old vichan. I put diff comments on them.

File: templates/themes/ukko/theme.php (hunk 1 above)
Comment (mgrabovsky) [inline templates/themes/ukko/theme.php]: Yes, see [templates/index.html:61](https://github.com/lainchan/lainchan/blob/4b40f69a4e6d190f6d342875bd084c2a6c9d8b0f/templates/index.html#L61). Or we could fix it there, but that may not be backwards-compatible.

File: templates/themes/catalog/info.php (hunk 2 above)
Comment (mgrabovsky) [inline templates/themes/catalog/info.php]: Sorry, I just have an urge to removing all trailing whitespace after opening a file.

File: inc/functions.php (hunk 3 above)
Comment (mgrabovsky) [inline inc/functions.php]: Yes, I get fatal errors on my install otherwise. This belongs to a different commit, but I rebased this one on top of that, so GitHub included it here.
